src/ert/config/summary_config.py

```diff
@@ -57,17 +57,31 @@ def read_from_file(self, run_path: str, iens: int) -> xr.Dataset:
     def response_type(self) -> str:
         return "summary"
 
+    @no_type_check
     @classmethod
     def from_config_dict(cls, config_dict: ConfigDict) -> Optional[SummaryConfig]:
         refcase = Refcase.from_config_dict(config_dict)
-        if summary_keys := config_dict.get(ConfigKeys.SUMMARY, []):  # type: ignore
-            eclbase: Optional[str] = config_dict.get("ECLBASE")  # type: ignore
+        if summary_keys := config_dict.get(ConfigKeys.SUMMARY, []):
+            eclbase: Optional[str] = config_dict.get("ECLBASE")
             if eclbase is None:
                 raise ConfigValidationError(
                     "In order to use summary responses, ECLBASE has to be set."
                 )
             time_map = set(refcase.dates) if refcase is not None else None
-
+            forward_model = config_dict.get(ConfigKeys.FORWARD_MODEL, [])
+            names = [step[0] for step in forward_model]
+            simulation_step_exists = False
+            for job_name in names:
```

Comment:
Would refactor this a bit to something like this:
```
simulation_step_exists = any(_name for _name in names if _name.lower() in ["eclipse","flow"])
```

and also not use `job_name` and stick with `fm_step_name` or `step_name`, starting w/ calling the `names` `fm_step_names`

Reply:
`simulation_step_exists = any(_name for _name in names if _name.lower() in ["eclipse","flow"])`

This looks nice and it is also equivalent to `simulation_step_exists = any(_name.lower() in ["eclipse","flow"]) for _name in names` but it does not capture the same logic. 

To do everything in one line it should be something like:

`any(any(sim_name in _name.lower() for sim_name in ["eclipse","flow"]) for _name in names)`